x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/retry/ErrorResponse.java

```diff
@@ -34,4 +34,16 @@ public String getErrorMessage() {
     public boolean errorStructureFound() {
         return errorStructureFound;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Needed these so the `assertThat` calls in the error parser tests can succeed.

Reply:
Yeah that would make it cleaner. At the moment `ErrorResponse` is a base class and extended in a few places so we'd have to switch that to composition (I don't think we can extend a record 🤔 ).